### PR TITLE
331 update readme

### LINKED ISSUE #331: Update README
**Is your feature request related to a problem? Please describe.**
The readme is starting to feel a bit dated. 

**Describe the solution you'd like**
I think we should:

- [x] Update the features list
- [x] Consider adding more of a tutorial
- [x] Update the contribution guide
- [x] Remove the Version numbering scheme information from here.
- [x] Probably do research on what good README's are.

### PR DESCRIPTION
**Pull request by MicahGale**
# Description

This is a revision of the README with feedback from @jpmorgan98 as part of the review of MontePy for [pyOpenSCi](https://github.com/pyOpenSci/software-submission/issues/205). 

This included:

* Updating example code to be runnable by `doctest`.
* Added required badges to be PyOpenSci compliant
* Added citation information (will need to update once JOSS article is complete). 
* Added comparisons to competition.
* Updated contributing guide for accuracy.

Currently the too much example code is broken to implement CI testing here, but that is currently in the works in #575. 

Fixes #331

# Checklist

- [x] I have made corresponding changes to the documentation (if applicable)

### CODE REVIEW

**Comment by MicahGale** [comment]
@jpmorgan98, if you are willing to do so, I would appreciate your review on this PR.

File: README.md [hunk 1]
```diff
@@ -40,41 +45,40 @@ and the Python API documentation.
...
+* Currently has over 550 test cases.
 
  
 Here is a quick example showing multiple tasks in MontePy:
 
 ```python
-import montepy
-# read in file
-problem = montepy.read_input("foo.imcnp")
+>>> import montepy
```
Comment (jpmorgan98) [inline README.md:54]: If the workflow of this example is in a python script I would recommend removing the `>>>` and `...` so folks can copy paste into files more easily

Comment (MicahGale) [reply README.md:54]: Right, that's a good point. This formatting is necessary for the code to be recognized by `doctest` to ensure it's tested. Though there are some viable alternatives possibly that I need to look into further. 

`ipython` is able to understand this notation, but obviously that wouldn't work for dropping this in a `.py` file.

**Review by jpmorgan98 — APPROVED** [review]

**Review by jpmorgan98 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (MicahGale) [reply README.md]: I found a better library, so this is now unnecessary.

**Comment by MicahGale** [comment]
image
The test is failing right now because nature.com is currently down. Wasn't expecting that to be the issue.

**Review by tjlaboss — APPROVED** [review]

**Pull request merged**